### PR TITLE
C: Support generics

### PR DESCRIPTION
**Pull request by Smit-create**

### CODE REVIEW

File: src/libasr/codegen/asr_to_c.cpp [hunk 1]
```diff
@@ -516,6 +516,9 @@ class ASRToCVisitor : public BaseCCPPVisitor<ASRToCVisitor>
                     sub = format_type_c("", "const " + const_underlying_type + " ",
                                         v.m_name, false, false);
                 }
+            } else if (ASR::is_a<ASR::TypeParameter_t>(*v_m_type)) {
```
Comment (czgdp1807) [inline src/libasr/codegen/asr_to_c.cpp:519]: If I am not wrong backend shouldn't need to care if there are any generics in the frontend? The code for each type combination should be generated by the `instantiate_template` pass? Correct @certik?

Comment (certik) [reply src/libasr/codegen/asr_to_c.cpp:519]: Yes, everything should be already instantiated, so the generics should work in all backends, no special logic needed.

Comment (czgdp1807) [reply src/libasr/codegen/asr_to_c.cpp:519]: So this PR should never have been needed. @Smit-create Can you check what's there in ASR which is causing you to handle generics in C backend? I guess ASR is not correct that's why you created this PR for C backend.

Comment (Smit-create) [reply src/libasr/codegen/asr_to_c.cpp:519]: Opened an issue for it's discussion: #1471

**Comment by certik** [comment]
Converted to a Draft for now.

File: src/libasr/codegen/asr_to_c.cpp (hunk 1 above)
Comment (Smit-create) [reply src/libasr/codegen/asr_to_c.cpp:519]: As per the discussion in #1471, the current ASR is fine and this diff fixes the codegen of generics in C backend.

**Review by czgdp1807 — APPROVED** [review]

**Pull request merged**